arcadia_pycolor/mpl.py

```diff
@@ -1,18 +1,334 @@
-from typing import TYPE_CHECKING
+import matplotlib as mpl
+import matplotlib.font_manager as font_manager
+import matplotlib.pyplot as plt
+from matplotlib import colormaps
+from matplotlib.legend import Legend
+from matplotlib.lines import Line2D
+from matplotlib.offsetbox import DrawingArea
 
-import matplotlib.colors as mcolors
+import arcadia_pycolor.colors as colors
+import arcadia_pycolor.gradients
+import arcadia_pycolor.palettes
+from arcadia_pycolor.palette import Palette
+from arcadia_pycolor.style_defaults import (
+    ARCADIA_RC_PARAMS,
+    DEFAULT_FONT,
+    FIGURE_PADDING,
+    FIGURE_SIZES,
+    FONT_FILTER,
+    MONOSPACE_FONT,
+)
 
-if TYPE_CHECKING:
-    from arcadia_pycolor.classes import Gradient, Palette
+LEGEND_PARAMS = dict(
+    alignment="left",
+    title_fontproperties={"weight": "semibold", "size": "26"},
+)
 
+SAVE_WEB = dict(dpi=72, bbox_inches="tight", pad_inches=0.41)
+SAVE_PRINT = dict(dpi=300, bbox_inches="tight", pad_inches=0.41)
 
-def palette_to_cmap(palette: "Palette"):
-    return mcolors.ListedColormap([color.hex_code for color in palette.colors], palette.name)
 
+def _find_axis(axis=None):
+    "Convenience function to catch the current axis if none is provided."
+    if axis is None:
+        return plt.gca()
+    else:
+        return axis
 
-def gradient_to_linear_cmap(gradient: "Gradient"):
-    colors = [(value, color.hex_code) for value, color in zip(gradient.values, gradient.colors)]
-    return mcolors.LinearSegmentedColormap.from_list(
-        gradient.name,
-        colors=colors,
+
+def set_yticklabel_fonts(axis=None, font: str = DEFAULT_FONT):
```

Comment:
Nit: there's only one font to set (and ditto for `set_xticklabel_fonts` and `set_ticklabel_fonts`)
```suggestion
def set_yticklabel_font(axis=None, font: str = DEFAULT_FONT):
```